test/gcs/gcs.go

```diff
@@ -34,54 +30,139 @@ import (
 	"google.golang.org/api/option"
 )
 
-var client *storage.Client
+type GCSClient struct {
+	client *storage.Client
+}
 
-// Authenticate explicitly sets up authentication for the rest of run
-func Authenticate(ctx context.Context, serviceAccount string) error {
-	var err error
-	client, err = storage.NewClient(ctx, option.WithCredentialsFile(serviceAccount))
-	return err
+// NewClient creates new GCS client with given service account
+func NewClient(ctx context.Context, serviceAccount string) (*GCSClient, error) {
+	client, err := storage.NewClient(ctx, option.WithCredentialsFile(serviceAccount))
+	if err != nil {
+		return nil, err
+	}
+	return &GCSClient{client: client}, nil
+}
+
+// StorageBucket creates a new bucket in GCS with uniform access policy
+func (g *GCSClient) NewStorageBucket(ctx context.Context, bucketName, project string) error {
```

Comment:
I think it would be nice to have that flexibility here. In case anyone ever wants to execute commands with different contexts.

Reply:
That would be nice, but could also be achieved by using a different client. Using a shared context is a pattern I have seen in multiple places, and I think it does help reducing quite a few code

Reply:
Sure I'll make that change.

Reply:
Actually...I think since the exposure is through the interface Client, we need to expose context since it is something any struct that implements the interface Client will need to have.